### PR TITLE
Add missing pluginlib xml files

### LINKED ISSUE #22: missing pluginlib xml files for each controller
With #19 the individual controllers can't be loaded anymore as the according `plugin.xml` file is missing. 

@jordan-palacios : How did you test to load the controllers in the controller manager? Maybe I am missing something here.

### PR DESCRIPTION
**Pull request by jordan-palacios**
Fixes #22.

### CODE REVIEW

**Comment by jordan-palacios** [comment]
@bmagyar @Karsten1987 it seems the `ros2_controllers` travis is running against the crystal version of `ros2_control`:
https://travis-ci.org/ros-controls/ros2_controllers/jobs/653481872#L316
```
=== src/ros-controls/ros2_control (git) ===
Cloning into '.'...
Switched to a new branch 'crystal'
Branch 'crystal' set up to track remote branch 'crystal' from 'origin'.
```

**Comment by Karsten1987** [comment]
@jordan-palacios I took the liberty to push on your branch to fix the additional travis repos. They were indeed still set to `crystal`. I further opened up a PR to not export the class loader header, which should avoid the `boost` error you're currently seeing in the travis output.

File: joint_state_controller/test/test_load_joint_state_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "controller_manager/controller_manager.hpp"
+
+#include "test_robot_hardware/test_robot_hardware.hpp"
+
+#include "rclcpp/utilities.hpp"
```
Comment (Karsten1987) [inline joint_state_controller/test/test_load_joint_state_controller.cpp]: nit: alphasort

File: joint_state_controller/test/test_load_joint_state_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    std::make_shared<test_robot_hardware::TestRobotHardware>();
+
+  robot->init();
+
+  std::shared_ptr<rclcpp::executor::Executor> executor =
+    std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+
+  controller_manager::ControllerManager cm(robot, executor, "test_controller_manager");
+
+  ASSERT_NO_THROW(cm.load_controller("test_joint_state_controller",
```
Comment (Karsten1987) [inline joint_state_controller/test/test_load_joint_state_controller.cpp]:
```suggestion
  ASSERT_NO_THROW(
    cm.load_controller(
      "test_joint_state_controller",
      "joint_state_controller/JointStateController"));
```

uncrustify (master) is complaining about this.

File: joint_state_controller/test/test_load_joint_state_controller.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "joint_state_controller/JointStateController"));
```
Comment (Karsten1987) [inline joint_state_controller/test/test_load_joint_state_controller.cpp]:
```suggestion
```

change connected to the suggestion above. Github didn't let us change multiline comments.

File: joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    std::make_shared<test_robot_hardware::TestRobotHardware>();
+
+  robot->init();
+
+  std::shared_ptr<rclcpp::executor::Executor> executor =
+    std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+
+  controller_manager::ControllerManager cm(robot, executor, "test_controller_manager");
+
+  ASSERT_NO_THROW(cm.load_controller("test_joint_trajectory_controller",
```
Comment (Karsten1987) [inline joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp]:
```suggestion
  ASSERT_NO_THROW(
    cm.load_controller(
      "test_joint_trajectory_controller",
      "joint_trajectory_controller/JointTrajectoryController"));
```

File: joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "joint_trajectory_controller/JointTrajectoryController"));
```
Comment (Karsten1987) [inline joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp]:
```suggestion
```

File: joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "controller_manager/controller_manager.hpp"
+
+#include "test_robot_hardware/test_robot_hardware.hpp"
+
+#include "rclcpp/utilities.hpp"
```
Comment (Karsten1987) [inline joint_trajectory_controller/test/test_load_joint_trajectory_controller.cpp]: alpha

File: joint_trajectory_controller/CMakeLists.txt [hunk 7]
```diff
@@ -74,6 +79,16 @@ if(BUILD_TESTING)
...
   )
+
+  ament_add_gtest(
+    test_load_joint_trajectory_controller
+    test/test_load_joint_trajectory_controller.cpp
+  )
+  target_include_directories(test_load_joint_trajectory_controller PRIVATE include)
+  ament_target_dependencies(test_load_joint_trajectory_controller
+    controller_manager
+    test_robot_hardware
```
Comment (ddengster) [inline joint_trajectory_controller/CMakeLists.txt]: I got include errors when building this with a tweaked example code shown in the ros2_control github. Not sure how it passed the automated tests?

Anyway, adding the following to ament_target_dependencies fixes it.
`"controller_interface"`

Comment (bmagyar) [reply joint_trajectory_controller/CMakeLists.txt]: The code here does not on it's own impose any dependencies on that library, perhaps one of the dependencies included does not properly link/export it and your local setup produced a broken tree due to it.

Comment (Karsten1987) [reply joint_trajectory_controller/CMakeLists.txt]: @ddengster Do you have this commit on your `ros2_control`? https://github.com/ros-controls/ros2_control/pull/58

Comment (ddengster) [reply joint_trajectory_controller/CMakeLists.txt]: > @ddengster Do you have this commit on your `ros2_control`? [ros-controls/ros2_control#58](https://github.com/ros-controls/ros2_control/pull/58)

Ah that fixed it, thanks.

**Comment by jordan-palacios** [comment]
@bmagyar @Karsten1987 Thanks for the fixes. I think this is ready to merge.

File: joint_trajectory_controller/CMakeLists.txt (hunk 7 above)
Comment (bmagyar) [reply joint_trajectory_controller/CMakeLists.txt]: If you guys can confirm it is a problem, please open new issue to track it.

**Pull request merged**